app/src/main/java/com/halim/adam/cashmaster/AddNewIncome.java

```diff
@@ -4,17 +4,35 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.EditText;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 
 public class AddNewIncome extends Activity {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_add_new_income);
     }
 
-    public void MoveToTransactionHistoryPage(View view){
+    public void MoveToTransactionHistoryPage(View view) throws ParseException {
         Intent intent = new Intent(this, TransactionHistory.class);
         startActivity(intent);
+
+        EditText inputName = findViewById(R.id.inputName);
+        EditText inputAmount = findViewById(R.id.inputAmount);
+        EditText inputDate = findViewById(R.id.inputDate);
+        Date date = new Date();
+
+        DatabaseHelper dbHelper = new DatabaseHelper(this);
+
+        if (!inputDate.getText().toString().matches("")){
```

Comment:
what happen if the inputDate is missing ?

Reply:
It will use the InsertIncome method with default values of the current date sir